### DISCUSSION TITLE
How do I use autofill?

### DISCUSSION THREAD

**Discussion by victormihalache** [Kunzisoft/KeePassDX#1279] [Q&A]
I don't know if this is a bug or what. I enable autofill on my samsung S10 with florisboard and nothing pops up. From all the screenshots I've seen there's something that pops up when selecting a password/username field... What do I have to do?

**Comment by J-Jamet** [comment]
You do not indicate which browser and which form you use and want to fill in.
Make sure you read the whole wiki.
https://github.com/Kunzisoft/KeePassDX/wiki/AutoFill

**Reply by victormihalache** [reply]
Oh yea sorry. it was DuckDuckGo on google. The wiki didn't help and I've done every setup there (bar the one for 'weird phones')

**Reply by J-Jamet** [reply]
In this case, try to restart your device, it may restart the Autofill service of it.

**Reply by victormihalache** [reply]
I restarted the device, and still didn't work on DDG, but it did show up on 3rd party apps such as google's native setup (similar to the one you see when first setting up a device). So either I have the wrong URI set up for keepass (likely) or there's a bug that makes it not show up there.

Quick note: when holding the text field and pressing "autofill" it doesn't do anything (at least in the browser). Will test other browsers later today.

**Reply by J-Jamet** [reply]
The autofill has nothing to do with the URI, it is the service of the device that manages it and the application provides it with the algorithm to recognize the form fields.
If you can see the popup "Sign in with KeePassDX", it's already a good point, it means that the service is connected to KeePassDX.

> Quick note: when holding the text field and pressing "autofill" it doesn't do anything (at least in the browser). Will test other browsers later today.

Maybe it's just that the form is not recognized. Spoiler alert: if there is no password field, it's unlikely to be recognized.

**Reply by victormihalache** [reply]
Well then that's the problem, since google does the password login on a different page.

**Reply by J-Jamet** [reply]
Related issue : https://github.com/Kunzisoft/KeePassDX/issues/1272

**Reply by ghost** [reply]
Not just Google. It seems to be a trend and it increases the chances to be phished when you have to select the entry manually each time. I've tried KeePass2Android and while it also struggles with the Auto-Fill service restrictions, it seems to be better at auto-filling fields.
